# CAPSTONE PROJECT : MONEY-TOR [DATA SCIENCE]
- CDCC325D6Y0761 - Ignatius William Alvin
- CDCC325D6Y2656 - Jason Nathannael

## PERTANYAAN BISNIS
1. Bagaimana tren total pendapatan transaksi UMKM dari tahun 2024 - 2026?

2. Kategori produk atau layanan apa yang menghasilkan total pendapatan tertinggi selama periode 2025 - 2026?

3. Bagaimana pengaruh event promosi terhadap nilai transaksi yang berhasil selama periode 2024 - 2026?

4. Metode pembayaran apa yang paling sering digunakan dan menghasilkan nilai transaksi terbesar pada tahun 2026?

## IMPORT PACKAGES YANG DIPERLUKAN

In [ ]:
# Import untuk Manipulasi Data
import pandas as pd
import numpy as np

# Import untuk Visualisasi Data
import matplotlib.pyplot as plt
import seaborn as sns

# Import untuk Tanggal
import datetime as dt

## DATA WRANGLING

### Gathering Data

In [ ]:
# Membaca Dataset yang digunakan
transactions_df = pd.read_csv('Transactions.csv')
sellers_df = pd.read_csv('Sellers.csv')
products_df = pd.read_csv('Products.csv')
suppliers_df = pd.read_csv('Suppliers.csv')
logs_df = pd.read_csv('Logs.csv')

### Insight Gathering Data

- Pada tahap gathering data, dilakukan proses pengumpulan data dari berbagai sumber file dalam format CSV menggunakan library pandas dengan fungsi `pd.read_csv()`.

- Dataset yang digunakan terdiri dari beberapa tabel yang merepresentasikan entitas dalam sistem UMKM, seperti data transaksi (Transactions), data penjual (Sellers), data produk (Products), data penyuplai (Suppliers), serta data pendukung berupa riwayat aktivitas pembelian (Logs).

- Secara keseluruhan, dataset telah terstruktur dengan baik dan siap digunakan pada tahap analisis berikutnya, yaitu assessing data dan cleaning data.

### Assessing Data

In [ ]:
# Mengecek ukuran setiap dataset menggunakan fungsi df.shape
print("Dataset Transactions:", transactions_df.shape)
print("Dataset Sellers:", sellers_df.shape)
print("Dataset Products:", products_df.shape)
print("Dataset Suppliers:", suppliers_df.shape)
print("Dataset Logs:", logs_df.shape)

Dataset Transactions: (30000, 14)
Dataset Sellers: (6, 5)
Dataset Products: (40, 11)
Dataset Suppliers: (8, 4)
Dataset Logs: (10000, 9)


In [ ]:
# Menampilkan informasi dari setiap dataset menggunakan fungsi df_info
transactions_df.info()
print("\n")

sellers_df.info()
print("\n")

products_df.info()
print("\n")

suppliers_df.info()
print("\n")

logs_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id_transaksi       30000 non-null  object
 1   id_seller          30000 non-null  object
 2   id_produk          30000 non-null  object
 3   tanggal            30000 non-null  object
 4   jam_transaksi      30000 non-null  object
 5   jenis_transaksi    30000 non-null  object
 6   kategori           30000 non-null  object
 7   qty                30000 non-null  int64 
 8   harga_satuan       30000 non-null  int64 
 9   total_harga        30000 non-null  int64 
 10  metode_pembayaran  30000 non-null  object
 11  event              30000 non-null  object
 12  diskon             30000 non-null  object
 13  status_transaksi   30000 non-null  object
dtypes: int64(3), object(11)
memory usage: 3.2+ MB


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6 entries, 0 to 5
Data columns (tota

In [ ]:
# Menampilkan 5 baris teratas dari dataset Transactions menggunakan fungsi df.head
transactions_df.head()

,id_transaksi,id_seller,id_produk,tanggal,jam_transaksi,jenis_transaksi,kategori,qty,harga_satuan,total_harga,metode_pembayaran,event,diskon,status_transaksi
0,T000001,U006,P040,2025-08-23,10:47,Pemasukan,Fashion,2,50000,100000,QRIS,Tidak Ada Event,Harga Normal,Berhasil
1,T000001,U006,P036,2025-08-23,10:47,Pemasukan,Fashion,1,150000,150000,QRIS,Tidak Ada Event,Harga Normal,Berhasil
2,T000002,U005,P032,2026-10-10,08:35,Pemasukan,Laundry,5,15000,75000,QRIS,10.10 Sale,Harga Normal,Berhasil
3,T000002,U005,P034,2026-10-10,08:35,Pemasukan,Laundry,1,7000,7000,QRIS,10.10 Sale,Harga Normal,Berhasil
4,T000003,U002,P007,2026-07-14,12:17,Pemasukan,Sembako,7,70000,416500,QRIS,Tidak Ada Event,Diskon 15%,Berhasil


In [ ]:
# Menampilkan 5 baris teratas dari dataset Sellers menggunakan fungsi df.head
sellers_df.head()

,id_seller,nama_usaha,jenis_usaha,lokasi_usaha,tanggal_bergabung
0,U001,Warkop GenZ,Kafe,Bandung,2024-01-10
1,U002,Warung Madura,Retail,Cimahi,2024-03-01
2,U003,Dapur Nusantara,Kuliner,Garut,2024-07-15
3,U004,Roti Bahagia,Bakery,Sumedang,2024-10-01
4,U005,CleanWash,Laundry,Bandung,2025-02-01


In [ ]:
# Menampilkan 5 baris teratas dari dataset Products menggunakan fungsi df.head
products_df.head()

,id_produk,id_seller,id_supplier,nama_produk,kategori_produk,harga_jual,harga_modal,stok_awal,minimum_stok,status_produk,tanggal_input_produk
0,P001,U001,S001,Americano,Minuman,15000,7000,757,151,aktif,2024-01-21
1,P002,U001,S001,Latte,Minuman,20000,10000,825,165,aktif,2024-01-15
2,P003,U001,S001,Kopi Susu,Minuman,18000,9000,752,150,aktif,2024-01-22
3,P004,U001,S007,Coca-cola,Minuman,10000,5000,744,149,aktif,2024-01-19
4,P005,U001,S007,Air Mineral,Minuman,5000,2500,715,143,aktif,2024-01-11


In [ ]:
# Menampilkan 5 baris teratas dari dataset Suppliers menggunakan fungsi df.head
suppliers_df.head()

,id_supplier,nama_supplier,kategori_supplier,lokasi
0,S001,PT Kopi Nusantara,Biji Kopi & Minuman,Bandung
1,S002,CV Sembako Jaya,Sembako & FMCG,Bekasi
2,S003,UD Dapur Nusantara,Bahan Makanan,Yogyakarta
3,S004,UD Bakery Fresh,Bakery & Snack,Jakarta
4,S005,CV Kebutuhan Rumah,Household & Laundry,Tangerang


In [ ]:
# Menampilkan 5 baris teratas dari dataset Logs menggunakan fungsi df.head
logs_df.head()

,id_log,id_produk,id_seller,tanggal,jenis_perubahan,jumlah,stok_sebelum,stok_sesudah,alasan
0,L001217,P005,U001,2024-01-10,Retur Supplier,13,6473,6486,Pengembalian supplier
1,L003688,P001,U001,2024-01-10,Penjualan,4,15514,15510,Produk terjual
2,L005182,P001,U001,2024-01-11,Restock,129,20843,20972,Pembelian supplier
3,L008456,P005,U001,2024-01-11,Restock,431,29209,29640,Pembelian supplier
4,L000833,P001,U001,2024-01-12,Penjualan,29,4576,4547,Produk terjual


In [ ]:
# Mengecek apakah ada duplikasi data pada setiap dataset menggunakan fungsi df.duplicated().sum()
print("Jumlah duplikasi data pada dataset Transactions:", transactions_df.duplicated().sum())
print("Jumlah duplikasi data pada dataset Sellers:", sellers_df.duplicated().sum())
print("Jumlah duplikasi data pada dataset Products:", products_df.duplicated().sum())
print("Jumlah duplikasi data pada dataset Suppliers:", suppliers_df.duplicated().sum())
print("Jumlah duplikasi data pada dataset Logs:", logs_df.duplicated().sum())

Jumlah duplikasi data pada dataset Transactions: 0
Jumlah duplikasi data pada dataset Sellers: 0
Jumlah duplikasi data pada dataset Products: 0
Jumlah duplikasi data pada dataset Suppliers: 0
Jumlah duplikasi data pada dataset Logs: 0


In [ ]:
# Mengecek apakah ada missing Value pada dataset Transactions menggunakan fungsi df.isnull().sum()
transactions_df.isnull().sum()

,0
id_transaksi,0
id_seller,0
id_produk,0
tanggal,0
jam_transaksi,0
jenis_transaksi,0
kategori,0
qty,0
harga_satuan,0
total_harga,0


In [ ]:
# Mengecek apakah ada missing Value pada dataset Sellers menggunakan fungsi df.isnull().sum()
sellers_df.isnull().sum()

,0
id_seller,0
nama_usaha,0
jenis_usaha,0
lokasi_usaha,0
tanggal_bergabung,0


In [ ]:
# Mengecek apakah ada missing Value pada dataset Products menggunakan fungsi df.isnull().sum()
products_df.isnull().sum()

,0
id_produk,0
id_seller,0
id_supplier,0
nama_produk,0
kategori_produk,0
harga_jual,0
harga_modal,0
stok_awal,0
minimum_stok,0
status_produk,0


In [ ]:
# Mengecek apakah ada missing Value pada dataset Suppliers menggunakan fungsi df.isnull().sum()
suppliers_df.isnull().sum()

,0
id_supplier,0
nama_supplier,0
kategori_supplier,0
lokasi,0


In [ ]:
# Mengecek apakah ada missing Value pada dataset Logs menggunakan fungsi df.isnull().sum()
logs_df.isnull().sum()

,0
id_log,0
id_produk,0
id_seller,0
tanggal,0
jenis_perubahan,0
jumlah,0
stok_sebelum,0
stok_sesudah,0
alasan,0


In [ ]:
# Melihat deskripsi dataset Transactions menggunakan df.describe()
transactions_df.describe()

,qty,harga_satuan,total_harga
count,30000.000000,30000.000000,30000.000000
mean,3.056167,31387.500000,74713.088333
std,1.970417,34111.246381,72235.634933
min,1.000000,5000.000000,4000.000000
25%,2.000000,12000.000000,30000.000000
50%,3.000000,18000.000000,56000.000000
75%,4.000000,28000.000000,90000.000000
max,10.000000,150000.000000,700000.000000


In [ ]:
# Melihat deskripsi dataset Sellers menggunakan df.describe()
sellers_df.describe()

,id_seller,nama_usaha,jenis_usaha,lokasi_usaha,tanggal_bergabung
count,6,6,6,6,6
unique,6,6,6,4,6
top,U001,Warkop GenZ,Kafe,Bandung,2024-01-10
freq,1,1,1,3,1


In [ ]:
# Melihat deskripsi dataset Products menggunakan df.describe()
products_df.describe()

,harga_jual,harga_modal,stok_awal,minimum_stok
count,40.000000,40.000000,40.000000,40.000000
mean,28825.000000,16512.500000,1532.550000,306.550000
std,30696.936705,19004.549894,1173.538785,234.722307
min,5000.000000,2500.000000,412.000000,82.000000
25%,14750.000000,7000.000000,736.750000,147.500000
50%,18000.000000,9500.000000,954.000000,191.000000
75%,25750.000000,14000.000000,2063.750000,413.000000
max,150000.000000,90000.000000,4294.000000,859.000000


In [ ]:
# Melihat deskripsi dataset Suppliers menggunakan df.describe()
suppliers_df.describe()

,id_supplier,nama_supplier,kategori_supplier,lokasi
count,8,8,8,8
unique,8,8,8,7
top,S001,PT Kopi Nusantara,Biji Kopi & Minuman,Bandung
freq,1,1,1,2


In [ ]:
# Melihat deskripsi dataset Logs menggunakan df.describe()
logs_df.describe()

,jumlah,stok_sebelum,stok_sesudah
count,10000.000000,10000.000000,10000.000000
mean,133.029000,16518.934500,16633.710900
std,154.595036,10953.753394,10958.378013
min,1.000000,279.000000,279.000000
25%,16.000000,7597.250000,7740.000000
50%,36.000000,14481.500000,14550.500000
75%,245.000000,24021.250000,24103.250000
max,500.000000,49263.000000,49263.000000


### Insight Assessing Data

- Berdasarkan hasil assessing data, seluruh dataset berhasil dimuat dengan baik dan memiliki struktur data yang cukup terorganisir untuk mendukung proses analisis lebih lanjut.

- Dataset yang digunakan terdiri dari lima entitas utama, yaitu `Transactions`, `Sellers`, `Products`, `Suppliers`, dan `Logs`. Dataset transaksi menjadi sumber data terbesar dengan 30.000 baris dan 14 kolom, sedangkan dataset log aktivitas mencatat 10.000 perubahan inventori, menunjukkan bahwa data telah berkembang menjadi skala operasional yang lebih besar dibanding sebelumnya.

- Hasil pemeriksaan struktur menggunakan `df.info()` menunjukkan bahwa seluruh dataset tidak memiliki kehilangan data (non-null lengkap pada seluruh atribut). Namun, masih ditemukan beberapa atribut waktu yang bertipe object, seperti `tanggal`, `jam_transaksi`, `tanggal_bergabung`, dan `tanggal_input_produk`, sehingga perlu dilakukan penyesuaian tipe data pada tahap cleaning.

- Berdasarkan hasil peninjauan awal menggunakan `df.head()`, dataset transaksi saat ini memiliki struktur yang lebih rinci karena satu `id_transaksi` dapat terdiri dari beberapa produk dalam satu transaksi. Hal ini menunjukkan bahwa data transaksi disimpan pada level detail transaksi (transaction detail / line item), sehingga pengulangan `id_transaksi` tidak menunjukkan adanya duplikasi data.

- Hasil pengecekan menggunakan `df.duplicated().sum()` menunjukkan bahwa tidak ditemukan data duplikat pada seluruh dataset. Dengan demikian, seluruh data dianggap unik dan tidak memerlukan proses penghapusan data duplikat.

- Pemeriksaan missing value menggunakan `df.isnull().sum()` menunjukkan bahwa seluruh atribut pada semua dataset memiliki nilai lengkap tanpa data kosong. Berbeda dengan versi dataset sebelumnya, atribut `event` dan `diskon` pada dataset transaksi telah tersedia secara penuh dan tidak lagi memerlukan proses imputasi data.

- Berdasarkan statistik deskriptif pada dataset transaksi, rata-rata jumlah pembelian berada pada kisaran 3 item per transaksi dengan rata-rata nilai transaksi sekitar 74 ribu rupiah. Nilai maksimum transaksi mencapai 700 ribu rupiah, menunjukkan adanya variasi nilai transaksi antar aktivitas penjualan.

- Pada dataset produk, rata-rata harga jual berada pada kisaran 28 ribu rupiah dengan rata-rata stok awal lebih dari 1.500 unit per produk, yang menunjukkan kapasitas inventori yang lebih besar dibanding versi dataset sebelumnya.

- Dataset log menunjukkan aktivitas inventori yang cukup tinggi dengan rata-rata perubahan stok sebesar 133 unit per aktivitas dan nilai stok yang dapat mencapai lebih dari 49 ribu unit, sehingga berpotensi digunakan untuk analisis manajemen persediaan dan operasional usaha.

- Secara keseluruhan, hasil assessing menunjukkan bahwa dataset telah memiliki kualitas data yang baik, lengkap, dan konsisten sehingga siap dilanjutkan ke tahap cleaning data sebelum dilakukan exploratory data analysis (EDA) dan visualisasi data.

### Cleaning Data

In [ ]:
# Mengubah tipe data tanggal dari object menjadi datetime! #


# Kolom yang akan dikonversi antara lain
transactions_df['tanggal'] = pd.to_datetime(transactions_df['tanggal'])
sellers_df['tanggal_bergabung'] = pd.to_datetime(sellers_df['tanggal_bergabung'])
products_df['tanggal_input_produk'] = pd.to_datetime(products_df['tanggal_input_produk'])
logs_df['tanggal'] = pd.to_datetime(logs_df['tanggal'])

# Mengecek salah satu dataset yang sudah diubah kolom objectnya
transactions_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   id_transaksi       30000 non-null  object        
 1   id_seller          30000 non-null  object        
 2   id_produk          30000 non-null  object        
 3   tanggal            30000 non-null  datetime64[ns]
 4   jam_transaksi      30000 non-null  object        
 5   jenis_transaksi    30000 non-null  object        
 6   kategori           30000 non-null  object        
 7   qty                30000 non-null  int64         
 8   harga_satuan       30000 non-null  int64         
 9   total_harga        30000 non-null  int64         
 10  metode_pembayaran  30000 non-null  object        
 11  event              30000 non-null  object        
 12  diskon             30000 non-null  object        
 13  status_transaksi   30000 non-null  object        
dtypes: dat

In [ ]:
# Menangani format kolom Diskon

# Menghapus teks pada kolom diskon
transactions_df['diskon'] = (
    transactions_df['diskon']
    .str.replace('Diskon ', '', regex=False)
    .str.replace('%', '', regex=False)
    .replace('Harga Normal', '0')
)

# Mengubah tipe data menjadi integer yang sebelumnya object
transactions_df['diskon'] = (
    transactions_df['diskon']
    .astype(int)
)

# Mengecek status kolom diskon pada dataset transactions
transactions_df[['diskon']].info()

# Melihat nilai unik kolom diskon setelah cleaning
print(transactions_df['diskon'].unique())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   diskon  30000 non-null  int64
dtypes: int64(1)
memory usage: 234.5 KB
[ 0 15  5 20 10]


In [ ]:
# Validasi konsistensi perhitungan total_harga

# Menghitung ulang total transaksi
cek_total = transactions_df[
    (
        (
            transactions_df['qty']
            * transactions_df['harga_satuan']
            * (1 - transactions_df['diskon'] / 100)
        ).round(0)
        != transactions_df['total_harga']
    )
]

# Menampilkan hasil
cek_total

,id_transaksi,id_seller,id_produk,tanggal,jam_transaksi,jenis_transaksi,kategori,qty,harga_satuan,total_harga,metode_pembayaran,event,diskon,status_transaksi


In [ ]:
# Mengecekk konsistensi Data Kategorikal

# memastikan tidak ada penulisan kategori yang berbeda
print("Metode Pembayaran:")
print(transactions_df['metode_pembayaran'].unique())

print("\nStatus Transaksi:")
print(transactions_df['status_transaksi'].unique())

print("\nJenis Transaksi:")
print(transactions_df['jenis_transaksi'].unique())

print("\nKategori:")
print(transactions_df['kategori'].unique())

print("\nEvent:")
print(transactions_df['event'].unique())

print("\nStatus Produk:")
print(products_df['status_produk'].unique())

print("\nJenis Perubahan:")
print(logs_df['jenis_perubahan'].unique())

Metode Pembayaran:
['QRIS' 'Transfer' 'Cash']

Status Transaksi:
['Berhasil' 'Dibatalkan']

Jenis Transaksi:
['Pemasukan' 'Pengeluaran']

Kategori:
['Fashion' 'Laundry' 'Sembako' 'Frozen Food' 'Minuman' 'Bakery' 'Snack'
 'Makanan']

Event:
['Tidak Ada Event' '10.10 Sale' 'Lebaran' 'Ramadan' '3.3 Sale' '8.8 Sale'
 'Tahun Baru' '7.7 Sale' '6.6 Sale' '5.5 Sale' '2.2 Sale' 'Natal'
 '11.11 Sale' '4.4 Sale' '9.9 Sale' '12.12 Sale']

Status Produk:
['aktif' 'stok_menipis']

Jenis Perubahan:
['Retur Supplier' 'Penjualan' 'Restock' 'Barang Rusak' 'Kadaluarsa'
 'Penyesuaian']


In [ ]:
# Validasi log perubahan stok

# Memastikan perubahan stok masuk akal

cek_penjualan = logs_df[
    (logs_df['jenis_perubahan'] == 'Penjualan') &
    (
        logs_df['stok_sebelum']
        - logs_df['jumlah']
        != logs_df['stok_sesudah']
    )
]

cek_penjualan

,id_log,id_produk,id_seller,tanggal,jenis_perubahan,jumlah,stok_sebelum,stok_sesudah,alasan


### Insight Cleaning Data

- Pada tahap cleaning data dilakukan konversi tipe data pada seluruh atribut bertipe tanggal agar dapat digunakan untuk analisis berbasis waktu dan tren transaksi.

- Kolom diskon yang sebelumnya berbentuk teks berhasil ditransformasikan menjadi format numerik sehingga mempermudah proses perhitungan dan visualisasi.

- Dilakukan validasi terhadap perhitungan `total_harga` menggunakan kombinasi `qty`, `harga_satuan`, dan `diskon`, dan tidak ditemukan ketidaksesuaian pada seluruh data transaksi.

- Validasi data kategorikal menunjukkan seluruh kategori telah konsisten tanpa adanya duplikasi nilai akibat perbedaan penulisan.

- Validasi log inventori pada aktivitas penjualan menunjukkan perubahan stok telah sesuai dengan jumlah barang yang tercatat, sehingga data log dinilai valid untuk digunakan pada tahap analisis.

- Secara keseluruhan, dataset telah memenuhi kualitas data yang baik karena:
  1.   Tidak terdapat missing value
  2.   Tidak terdapat data duplikat
  3.   Format data sudah seragam
  4.   Hubungan antar variabel telah tervalidasi.

## Exploratory Data Analysis (EDA)

### Hal yang diexplore adalah:
1. Distribusi jumlah transaksi berdasarkan status transaksi
2. Distribusi transaksi berdasarkan kategori transaksi
3. Distribusi metode pembayaran yang digunakan
4. Statistik transaksi (kuantitas, harga satuan, dan total transaksi)
5. Distribusi transaksi berdasarkan waktu (tahunan dan bulanan)
6. Distribusi produk berdasarkan kategori dan status produk
7. Statistik harga dan stok produk
8. Karakteristik seller berdasarkan jenis usaha, lokasi, dan periode bergabung
9. Karakteristik supplier berdasarkan kategori dan lokasi supplier
10. Distribusi aktivitas inventori berdasarkan data log serta perubahan stok

Pada tahap ini dilakukan eksplorasi awal terhadap dataset UMKM untuk memahami karakteristik data `transaksi`, `produk`, `penjual`, `supplier`, dan aktivitas `inventori` selama periode 2024 - 2026. Eksplorasi dilakukan melalui pengamatan distribusi transaksi, metode pembayaran, pola waktu transaksi, karakteristik produk, kondisi stok, karakteristik seller dan supplier, serta aktivitas perubahan inventori.

Hasil eksplorasi ini digunakan untuk memperoleh pemahaman awal terhadap pola operasional UMKM dan hubungan antar data sebelum memasuki tahap Visualization & Explanatory Analysis untuk menjawab pertanyaan bisnis yang telah ditentukan.

In [ ]:
# Distribusi & Statistik Transaksi

print("===== Status Transaksi =====")
display(
    transactions_df['status_transaksi']
    .value_counts()
)

print("\n===== Kategori Transaksi =====")
display(
    transactions_df['kategori']
    .value_counts()
)

print("\n===== Metode Pembayaran =====")
display(
    transactions_df['metode_pembayaran']
    .value_counts()
)

print("\n===== Statistik Transaksi =====")
display(
    transactions_df[
        [
            'qty',
            'harga_satuan',
            'total_harga'
        ]
    ]
    .describe()
)

===== Status Transaksi =====


,count
status_transaksi,
Berhasil,29322
Dibatalkan,678



===== Kategori Transaksi =====


,count
kategori,
Fashion,5136
Bakery,5044
Minuman,5009
Laundry,4873
Snack,2575
Sembako,2518
Frozen Food,2425
Makanan,2420



===== Metode Pembayaran =====


,count
metode_pembayaran,
QRIS,16467
Cash,10594
Transfer,2939



===== Statistik Transaksi =====


,qty,harga_satuan,total_harga
count,30000.000000,30000.000000,30000.000000
mean,3.056167,31387.500000,74713.088333
std,1.970417,34111.246381,72235.634933
min,1.000000,5000.000000,4000.000000
25%,2.000000,12000.000000,30000.000000
50%,3.000000,18000.000000,56000.000000
75%,4.000000,28000.000000,90000.000000
max,10.000000,150000.000000,700000.000000


In [ ]:
# Distribusi Waktu Transaksi

transactions_df['tahun'] = (
    transactions_df['tanggal']
    .dt.year
)

transactions_df['bulan'] = (
    transactions_df['tanggal']
    .dt.month
)

print("===== Distribusi Tahunan =====")
display(
    transactions_df['tahun']
    .value_counts()
    .sort_index()
)

print("\n===== Distribusi Bulanan =====")
display(
    transactions_df['bulan']
    .value_counts()
    .sort_index()
)

===== Distribusi Tahunan =====


,count
tahun,
2024,4596
2025,12185
2026,13219



===== Distribusi Bulanan =====


,count
bulan,
1,1906
2,1969
3,2158
4,2235
5,2650
6,2386
7,2684
8,2719
9,2583


In [ ]:
# Eksplorasi Kategori, Harga, dan Stok Produk

print("===== Distribusi Kategori Produk =====")
display(
    products_df['kategori_produk']
    .value_counts()
)

print("\n===== Distribusi Status Produk =====")
display(
    products_df['status_produk']
    .value_counts()
)

print("\n===== Statistik Harga dan Stok Produk =====")
display(
    products_df[
        [
            'harga_jual',
            'harga_modal',
            'stok_awal',
            'minimum_stok'
        ]
    ].describe()
)

print("\n===== 5 Produk dengan Harga Jual Tertinggi =====")
display(
    products_df[
        [
            'nama_produk',
            'kategori_produk',
            'harga_jual'
        ]
    ]
    .sort_values(
        by='harga_jual',
        ascending=False
    )
    .head()
)

print("\n===== 5 Produk dengan Stok Awal Tertinggi =====")
display(
    products_df[
        [
            'nama_produk',
            'kategori_produk',
            'stok_awal'
        ]
    ]
    .sort_values(
        by='stok_awal',
        ascending=False
    )
    .head()
)

===== Distribusi Kategori Produk =====


,count
kategori_produk,
Minuman,5
Sembako,5
Snack,5
Makanan,5
Frozen Food,5
Bakery,5
Laundry,5
Fashion,5



===== Distribusi Status Produk =====


,count
status_produk,
aktif,37
stok_menipis,3



===== Statistik Harga dan Stok Produk =====


,harga_jual,harga_modal,stok_awal,minimum_stok
count,40.000000,40.000000,40.000000,40.000000
mean,28825.000000,16512.500000,1532.550000,306.550000
std,30696.936705,19004.549894,1173.538785,234.722307
min,5000.000000,2500.000000,412.000000,82.000000
25%,14750.000000,7000.000000,736.750000,147.500000
50%,18000.000000,9500.000000,954.000000,191.000000
75%,25750.000000,14000.000000,2063.750000,413.000000
max,150000.000000,90000.000000,4294.000000,859.000000



===== 5 Produk dengan Harga Jual Tertinggi =====


,nama_produk,kategori_produk,harga_jual
35,Jaket Kulit,Fashion,150000
38,Celana Cargo,Fashion,120000
36,Hoodie,Fashion,90000
37,Kaos Polo,Fashion,75000
6,Beras,Sembako,70000



===== 5 Produk dengan Stok Awal Tertinggi =====


,nama_produk,kategori_produk,stok_awal
33,Cuci Kiloan,Laundry,4294
31,Pemutih,Laundry,4148
34,Dry Cleaning,Laundry,3842
8,Telur Ayam,Sembako,3706
7,Gula,Sembako,3616


In [ ]:
# Eksplorasi Karakteristik Seller
print("===== Distribusi Jenis Usaha =====")
display(
    sellers_df['jenis_usaha']
    .value_counts()
)

print("\n===== Distribusi Lokasi Usaha =====")
display(
    sellers_df['lokasi_usaha']
    .value_counts()
)

print("\n===== Periode Bergabung Seller =====")
display(
    sellers_df[
        [
            'nama_usaha',
            'tanggal_bergabung'
        ]
    ]
    .sort_values(
        by='tanggal_bergabung'
    )
)

print("\n===== Jumlah Seller =====")
print(
    sellers_df['id_seller']
    .nunique()
)

===== Distribusi Jenis Usaha =====


,count
jenis_usaha,
Kafe,1
Retail,1
Kuliner,1
Bakery,1
Laundry,1
Fashion,1



===== Distribusi Lokasi Usaha =====


,count
lokasi_usaha,
Bandung,3
Cimahi,1
Garut,1
Sumedang,1



===== Periode Bergabung Seller =====


,nama_usaha,tanggal_bergabung
0,Warkop GenZ,2024-01-10
1,Warung Madura,2024-03-01
2,Dapur Nusantara,2024-07-15
3,Roti Bahagia,2024-10-01
4,CleanWash,2025-02-01
5,Thrift Hub,2025-05-01



===== Jumlah Seller =====
6


In [ ]:
# Eksplorasi Aktivitas Inventori

print("===== Distribusi Jenis Perubahan =====")
display(
    logs_df['jenis_perubahan']
    .value_counts()
)

print("\n===== Statistik Aktivitas Stok =====")
display(
    logs_df[
        [
            'jumlah',
            'stok_sebelum',
            'stok_sesudah'
        ]
    ].describe()
)

print("\n===== 5 Aktivitas dengan Perubahan Terbesar =====")
display(
    logs_df[
        [
            'id_produk',
            'jenis_perubahan',
            'jumlah'
        ]
    ]
    .sort_values(
        by='jumlah',
        ascending=False
    )
    .head()
)

print("\n===== 5 Stok Sesudah Tertinggi =====")
display(
    logs_df[
        [
            'id_produk',
            'stok_sesudah'
        ]
    ]
    .sort_values(
        by='stok_sesudah',
        ascending=False
    )
    .head()
)

===== Distribusi Jenis Perubahan =====


,count
jenis_perubahan,
Restock,4449
Penjualan,3557
Penyesuaian,677
Retur Supplier,526
Barang Rusak,497
Kadaluarsa,294



===== Statistik Aktivitas Stok =====


,jumlah,stok_sebelum,stok_sesudah
count,10000.000000,10000.000000,10000.000000
mean,133.029000,16518.934500,16633.710900
std,154.595036,10953.753394,10958.378013
min,1.000000,279.000000,279.000000
25%,16.000000,7597.250000,7740.000000
50%,36.000000,14481.500000,14550.500000
75%,245.000000,24021.250000,24103.250000
max,500.000000,49263.000000,49263.000000



===== 5 Aktivitas dengan Perubahan Terbesar =====


,id_produk,jenis_perubahan,jumlah
7185,P020,Restock,500
2514,P028,Restock,500
8001,P032,Restock,500
9825,P003,Restock,500
8083,P031,Restock,500



===== 5 Stok Sesudah Tertinggi =====


,id_produk,stok_sesudah
4702,P039,49263
5455,P039,49250
8945,P039,49246
5259,P039,48906
4394,P039,48859


In [ ]:
# Eksplorasi Supplier

print("===== Distribusi Kategori Supplier =====")
display(
    suppliers_df['kategori_supplier']
    .value_counts()
)

print("\n===== Distribusi Lokasi Supplier ===")
display(
    suppliers_df['lokasi']
    .value_counts()
)

print("\n===== Daftar Supplier =====")
display(
    suppliers_df[
        [
            'nama_supplier',
            'kategori_supplier',
            'lokasi'
        ]
    ]
)

print("\n===== Jumlah Supplier =====")
print(
    suppliers_df['id_supplier']
    .nunique()
)

===== Distribusi Kategori Supplier =====


,count
kategori_supplier,
Biji Kopi & Minuman,1
Sembako & FMCG,1
Bahan Makanan,1
Bakery & Snack,1
Household & Laundry,1
Fashion & Apparel,1
Minuman,1
Frozen Food,1



===== Distribusi Lokasi Supplier ===


,count
lokasi,
Bandung,2
Bekasi,1
Yogyakarta,1
Jakarta,1
Tangerang,1
Bogor,1
Malang,1



===== Daftar Supplier =====


,nama_supplier,kategori_supplier,lokasi
0,PT Kopi Nusantara,Biji Kopi & Minuman,Bandung
1,CV Sembako Jaya,Sembako & FMCG,Bekasi
2,UD Dapur Nusantara,Bahan Makanan,Yogyakarta
3,UD Bakery Fresh,Bakery & Snack,Jakarta
4,CV Kebutuhan Rumah,Household & Laundry,Tangerang
5,PT Fashion Supply Indonesia,Fashion & Apparel,Bandung
6,PT Beverage Supply,Minuman,Bogor
7,UD Frozen Food Nusantara,Frozen Food,Malang



===== Jumlah Supplier =====
8


## Insight Exploratory Data Analysis (EDA)

- Sebagian besar transaksi berada pada status Berhasil (29.322 transaksi), sedangkan transaksi Dibatalkan hanya 678 transaksi. Hal ini menunjukkan bahwa mayoritas aktivitas operasional pada dataset berjalan dengan baik.

- Kategori Fashion, Bakery, Minuman, dan Laundry menjadi kategori dengan jumlah transaksi tertinggi, sementara kategori lainnya memiliki aktivitas yang lebih rendah. Temuan ini menunjukkan distribusi transaksi belum merata pada seluruh kategori.

- Metode pembayaran QRIS memiliki frekuensi penggunaan tertinggi dibandingkan Cash dan Transfer. Hal ini menunjukkan adanya dominasi penggunaan pembayaran digital pada transaksi UMKM.

- Rata-rata transaksi terdiri dari sekitar 3 item dengan nilai rata-rata sekitar Rp74 ribu. Sebagian besar transaksi berada pada kelompok nilai menengah dengan variasi nilai yang cukup besar.

- Jumlah transaksi menunjukkan tren peningkatan selama periode 202 - 2026. Selain itu, aktivitas transaksi cenderung lebih tinggi pada bulan-bulan akhir tahun.

- Produk tersebar secara merata pada setiap kategori dan mayoritas masih berstatus aktif. Namun, terdapat beberapa produk dengan kondisi stok menipis yang perlu diperhatikan.

- Dataset mencakup berbagai jenis usaha dan kategori supplier dengan persebaran lokasi yang cukup beragam. Wilayah Bandung menjadi lokasi yang paling dominan.

- Perubahan stok paling banyak berasal dari aktivitas Restock dan Penjualan. Hal ini menunjukkan pengelolaan inventori aktif selama periode pengamatan.

## Visualization & Explanatory Analysis

### Pertanyaan bisnis 1: Bagaimana tren total pendapatan transaksi UMKM dari tahun 2024 - 2026?

In [ ]:
# kode pertanyaan bisnis 1

### Insight pertanyaan bisnis 1
- a
- b

### Pertanyaan bisnis 2: Kategori produk atau layanan apa yang menghasilkan total pendapatan tertinggi selama periode 2025 - 2026?

In [ ]:
# kode pertanyaan bisnis 2

### Insight pertanyaan bisnis 2
- a
- b

### Pertanyaan bisnis 3: Bagaimana pengaruh event promosi terhadap nilai transaksi yang berhasil selama periode 2024 - 2026?

In [ ]:
# kode pertanyaan bisnis 3

### Insight pertanyaan bisnis 3
- a
- b

### Pertanyaan bisnis 4: Metode pembayaran apa yang paling sering digunakan dan menghasilkan nilai transaksi terbesar pada tahun 2026?

In [ ]:
# kode pertanyaan bisnis 4

### Insight pertanyaan bisnis 4
- a
- b

## Analisis Lanjutan

In [ ]:
# kode untuk membuat file untuk keperluan dashboard streamlit

## KESIMPULAN PROSES ANALISA

1. a
2. b
3. c